blocks/rich-text/index.js

```diff
@@ -878,17 +878,35 @@ RichText.defaultProps = {
 };
 
 const RichTextContainer = compose( [
-	withBlockEditContext( ( { isSelected } ) => {
+	withInstanceId,
+	withBlockEditContext( ( context, ownProps ) => {
+		// When explicitly set as not selected, do nothing.
+		if ( ownProps.isSelected === false ) {
+			return {};
+		}
+		// When explicitly set as selected or has onFocus prop provided,
+		// use value stored in the context instead.
+		if ( ownProps.isSelected === true || ownProps.onFocus ) {
+			return {
+				isSelected: context.isSelected,
+			};
+		}
+		// Ensures that only one RichText component can be focused.
 		return {
-			isBlockSelected: isSelected,
+			isSelected: context.isSelected && context.focusedElement === ownProps.instanceId,
+			onSetup: () => {
```

Comment:
Is this going to incur a lot of unnecessary renders due to the new function references being passed each time (bypassing any pure component behavior)?

Reply:
There are also multiple props present  that change frequently causing rerenders. I tried to add some optimizations to avoid creating new references, but it wasn’t helpful at all. I didn’t notice any improvements

Reply:
"If you can't beat them, join them" ? 😆

Reply:
It would be great to take a closer look at all updates triggered by RichText and better understand how and why it works this way.